packages/editor/src/extensions/task-list/task-list.ts

```diff
@@ -49,7 +53,15 @@ export const TaskListNode = TaskList.extend({
     return {
       stats: {
         default: { checked: 0, total: 0 },
-        rendered: false
+        rendered: false,
+        parseHTML: (element) => {
+          if (!element.dataset.title) return { checked: 0, total: 0 };
+          const items = element.querySelectorAll("li.checklist--item");
+          const checked = Array.from(items).filter((item) =>
+            item.classList.contains("checked")
+          ).length;
+          return { checked, total: items.length };
+        }
```

Comment:
I like this. We can remove the logic in https://github.com/streetwriters/notesnook/blob/0921fef267a19a971c4e6a989500d6d0bd4db3c4/packages/editor/src/extensions/task-list/task-list.ts#L225-L244 which is supposed to do the same thing.